api/repository/delegate.ts

```diff
@@ -0,0 +1,84 @@
+import { Delegate } from 'shared/model/delegate';
+import Response from 'shared/model/response';
+import {Account} from 'shared/model/account';
+import { Transaction } from 'shared/model/transaction';
+
+// TODO need to instance Block
+declare class Block {}
+
+interface IForkDelegate {
+    delegatePublicKey: string;
+    blockTimestamp: bigint;
+    blockId: string;
+    blockHeight: number;
+    previousBlock: string;
+    cause: string;
+}
+
+export interface IDelegateRepository {
+    count(): Promise<Response<{ count: number }>>;
+
+    search(q: string, limit: number, sortField: string, sortMethod: string):
+        Promise<Response<{delegates: Delegate[]}>>;
```

Comment:
@Verbalist should repo return not 'clear' types?

Reply:
yes, Promise correct